debbuild

```diff
@@ -414,7 +460,7 @@ sub parse_cmd {
     my $opt_arg = shift;
     if ($cmdopts{type} eq 'b' or $cmdopts{type} eq 'd') {
       # Spec file
-      $specglobals{specfile} = $opt_arg;
+      $specglobals{_specfile} = $specglobals{specfile} = $opt_arg;
```

Comment:
This variable does not exist in RPM, so it will _not_ be defined in debbuild.

See https://github.com/rpm-software-management/rpm/pull/202

Reply:
`python-rpm-macros` is using this macro:
https://github.com/openSUSE/python-rpm-macros/blob/master/macros.lua#L56

Reply:
They broke compatibility with upstream RPM, which is what we target compatibility for. There was some commented out logic for this, but it was removed: https://github.com/openSUSE/python-rpm-macros/commit/296eb32e76011293f6949392aec8c12c63febdc9

Reply:
@Conan-Kudo considering debbuild is used heavily in OBS, hence by SUSE operating systems, `_specfile` is a reality we need to live with. IMHO we cannot just ignore it, no matter if we like it or not.

Reply:
That's a very close-minded way to look at it, especially when I just pointed out that python-rpm-macros had a fallback ripped out of its code for the absence of the macro.

Reply:
Actually, I think it is a very open-minded way to look at it, it's exactly the Robustness Principle by Postel: "be conservative in what you do, be liberal in what you accept from others".

https://en.wikipedia.org/wiki/Robustness_principle

Reply:
But debbuild hasn't supported Lua macros before, and me (along with the previous maintainers) have made it a point to align with upstream RPM. Moreover, in your case, upstream RPM has _another_ macro for this: https://github.com/rpm-software-management/rpm/pull/708#issuecomment-493405217

It was added two years ago. What this says to me is that the folks maintaining these macros are doing a bad job of following up with upstream after they explicitly pinged to say what they wanted was implemented.

Reply:
> What this says to me is that the folks maintaining these macros are doing a bad job of following up with upstream after they explicitly pinged to say what they wanted was implemented.

I'm not denying that, in fact I fully agree with that. But if those (deprecated and never standard) macros keep being used in the packages we need to use, I can only see three choices:

a. We fork the RPM packages and fix the usage of those macros. This is a lot of work and we may never succeed in getting those RPMs to get rid of non-standard macros.
b. Debbuild adds support for those macros, even if is advises against using them. That's what this PR is about. We could maybe add a warning so that when those outdated macros are used, debbuild shows a message like "this .spec file is using a non-standard macro, please ask the maintainer to fix that". Would that work for you?
c. We fork debbuild. Again, quite some work and something I would not like to do.

Reply:
Option A is relatively easy, since you folks are the upstream for your own macros. Send a change to them to make it use the standard macro and fallback to the SUSE-specific one when it doesn't exist (like in SLE < 15.3).

Reply:
> Option A is relatively easy, since you folks are the upstream for your own macros. Send a change to them to make it use the standard macro and fallback to the SUSE-specific one when it doesn't exist (like in SLE).

Problem is the 80+ packages (dependencies) we need in order to build Salt. We are not our own upstream in all the cases, and even when we (SUSE) are, it does not mean convincing a packager is an easy task.

Reply:
Sure, but each of those packages are not necessarily screwing around with self-referential macros like that. Those are all centralized in the crazy `%python_subpackages` macro in python-rpm-macros. So updating python-rpm-macros is enough to make things work as intended.

Reply:
In case if `specfile` is there we can put the extra macro definition like `%define _specfile %{specfile}` to make it working the same way, it just only adds extra call for expandmacros in runtime. I'm ok to remove this tweak, but it will push someone to make such tweaks in the projects. I supposed the main goal of `debbuild` is to allow building the packages with `spec` as much transparent and not to force the people to rewrite the specs (for sure in some cases we need to add extra checks to aling for the particular distro). In this case I would like to have the tweak I removed before to read the macro files from rpmbuild, I removed before, that tweak was more useful than this one.

Reply:
Hmm, on second thought, I guess we can keep it. At some point I'll need to refactor this to replace it with `%__file_name`.